miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/AjaxListSectionTag.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2002-2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.bbsrc.tgac.miso.webapp.util.form;
+
+import org.springframework.web.servlet.tags.RequestContextAwareTag;
+
+@SuppressWarnings("serial")
+public class AjaxListSectionTag extends RequestContextAwareTag {
+  private static long uniqueId;
+  private String config;
+
+  private String name;
+
+  private String target;
+
+  @Override
+  protected int doStartTagInternal() throws Exception {
+    pageContext.getOut().append(String.format(
+        "<br/><h1>%2$s</h1><table id='ajaxlist%1$d' class='display no-border ui-widget-content'></table><script type='text/javascript'>jQuery(document).ready(function () { ListUtils.createTable('ajaxlist%1$d', ListTarget.%3$s, null, %4$s);});</script>",
+        uniqueId++, name, target, config));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)

Reply:
:+1:

Reply:
Fixed.

Reply:
It's still doing the same thing, but in more lines. Can we just specify an ID in the JSP instead?

```
<miso:list-section-ajax id="statusAjaxlist" name="Order Status" target="completion" config="{ poolId: ${pool.id} }"/>
```

Reply:
I am very strongly opposed to this. No other tags work like this. The IDs can't be easily consumed on the client side because of the jQuery data tables.

Reply:
We previously had to provide IDs for these tables. Why is continuing to have to provide IDs a problem? It makes testing easier and doesn't involve this odd use of a static field

Reply:
Fixed.

Reply:
Fixed in other PR....